From 258cf5cc1715ea6dd523e54dbf221e20b940ccfb Mon Sep 17 00:00:00 2001
From: tracyone <tracyone@live.cn>
Date: Sun, 6 May 2018 19:07:53 +0800
Subject: [PATCH] =?UTF-8?q?=E6=94=AF=E6=8C=81=E6=98=BE=E7=A4=BAvmess?=
 =?UTF-8?q?=E9=93=BE=E6=8E=A5?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

vmess://
---
 serverinfo.py | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/serverinfo.py b/serverinfo.py
index 5b4f081..354ebd8 100755
--- a/serverinfo.py
+++ b/serverinfo.py
@@ -41,3 +41,31 @@ if readjson.ConfigDynPortRange:
     print("动态端口范围:%s") % str(readjson.ConfigDynPortRange)
 else:
     print("动态端口:禁止")
+
+config = {
+    "add": "",
+    "port": "",
+    "id": "",
+    "aid": "",
+    "net": "",
+    "type": "",
+    "tls": "",
+    "host": ""
+}
+
+config["add"] = str(readjson.ConfIP)
+config["port"] = str(readjson.ConfPort)
+config["id"] = str(readjson.ConfUUID)
+config["aid"] = str(readjson.ConfAlterId)
+config["net"] = str(readjson.ConfStreamNetwork)
+if readjson.ConfStreamNetwork == "kcp":
+    config["type"] = str(readjson.ConfStreamHeader)
+if (readjson.ConfStreamSecurity == "tls"):
+    config["tls"] = "tls"
+config["host"] = str(readjson.ConfPath)
+base64Str = base64.encodestring(json.dumps(config))
+base64Str = ''.join(base64Str.split())
+#green show
+print("\033[32m")
+print("vmess://%s") % base64Str
+print("\033[0m")
-- 
2.16.1

